### ISSUE TITLE
ohai report TUN interface as 'privateaddress'

### ISSUE THREAD

**Issue by bixu** [chef/ohai#491]
This even when the TUN interface comes after the 'real' private NIC:

```
$ ifconfig 
eth0      Link encap:Ethernet  HWaddr 90:b8:d0:45:8a:f7  
          inet addr:37.153.97.138  Bcast:37.153.97.255  Mask:255.255.254.0
          inet6 addr: fe80::92b8:d0ff:fe45:8af7/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:53735 errors:0 dropped:0 overruns:0 frame:0
          TX packets:29109 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:61487477 (61.4 MB)  TX bytes:3688564 (3.6 MB)

eth1      Link encap:Ethernet  HWaddr 90:b8:d0:7d:28:d1  
          inet addr:10.224.27.121  Bcast:10.224.27.255  Mask:255.255.252.0
          inet6 addr: fe80::92b8:d0ff:fe7d:28d1/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:422 errors:0 dropped:0 overruns:0 frame:0
          TX packets:4 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:24722 (24.7 KB)  TX bytes:852 (852.0 B)

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:16436  Metric:1
          RX packets:24 errors:0 dropped:0 overruns:0 frame:0
          TX packets:24 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:1200 (1.2 KB)  TX bytes:1200 (1.2 KB)

tun0      Link encap:UNSPEC  HWaddr 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00  
          inet addr:10.8.0.1  P-t-P:10.8.0.2  Mask:255.255.255.255
          UP POINTOPOINT RUNNING NOARP MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:100 
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

$ ohai | grep privateaddress
  "privateaddress": "10.8.0.1",
```

**Comment by lamont-granquist** [comment]
this does smell a bit wrong.  however, the algorithm for "what is my private address" without knowing any other details, will always fail in some use cases, because there is no standard that systems engineers agree on.

whatever we wind up doing this is probably ripe for making it configurable/hintable.

**Comment by tas50** [comment]
This has been fixed on master and will be shipping in 8.22.0 next week.
